hadoop-crypto/src/main/java/com/palantir/crypto2/hadoop/StandaloneEncryptedFileSystem.java

```diff
@@ -129,6 +131,14 @@ public boolean exists(Path path) throws IOException {
         return fs.exists(path);
     }
 
+    @Override
+    public boolean delete(Path path, boolean recursive) throws IOException {
+        // EncryptedFileSystem does not handle recursive deletes because it needs to rely on the KeyStorageStrategy.
+        // Here we know we are using the FileKeyStorageStrategy and can therefore delegate deletes to the FileSystem
+        // which will remove both the data and key material files.
+        return delegate.delete(path, recursive);
```

Comment:
what if path is a file rather than a folder?

Reply:
the key material won't get deleted, so this seems broken

Reply:
fixed in the follow up